amlnx_adapter/amlnx_adapter/arangoDB_networkx_arango_adapter.py

```diff
@@ -0,0 +1,178 @@
+        }
+        
+        for key, value in graph_params.items():
+            if key == gp.DB_DATA_QUERY:
+                self.cfg[gp.DATA_SECTION][key] = value
+            else:
+                self.cfg[gp.DB_SECTION][key] = value
+        return  
+    
+    def is_valid_config(self, graph_config):
+        gp = GraphParams()
+        valid_config = True
+        
+        if not gp.DB_SERVICE_HOST in graph_config:
+            print("managed service host is missing in supplied config")
+            valid_config = False
+        if not gp.DB_NAME in graph_config:
+            print("DB name is missing in supplied config")
+            valid_config = False
+        if not gp.DB_SERVICE_PORT in graph_config:
+            print("managed service port is missing in supplied config, using 8529")
+            graph_config[gp.DB_SERVICE_PORT] = 8529
+        if not gp.DB_USER_NAME in graph_config:
+            print("User name is missing in supplied config")
+            valid_config = False
+        if not gp.DB_PASSWORD in graph_config:
+            print("User password is not in supplied config")
+            valid_config = False
+        if not gp.DB_CONN_PROTOCOL in graph_config:
+            print("conn protocol is not in supplied config, using https")
+            graph_config[gp.DB_CONN_PROTOCOL] = 'https'
+        if not gp.DB_DATA_QUERY in graph_config:
+            print("query to load data is not provided")
+            valid_config = False
+        if not gp.DB_VERTEX_COL in graph_config:
+            print("vertex names in schema are not provided")
+            valid_config = False
+        if not gp.DB_EDGE_COL in graph_config:
+            print("edge names in schema is not provided")
+            valid_config = False
+
+        return valid_config
+    
+    def create_networkx_graph(self, graph_def = 'FraudDetection'):
+        self.set_db_connection()
+        gd = GraphParams()
+        
+        #query = self.cfg['queries']['load_data']
+                #get vertex names
+        g = nx.DiGraph()
+        for k,v in  self.cfg[gd.DB_SECTION][gd.DB_VERTEX_COL].items():
+            query = "FOR doc in %s " % (k)
+            cspl = [s +':'+ 'doc.' + s for s in v]
+            cspl.append('_id: doc._id')
+            csps = ','.join(cspl)
+            query = query + "RETURN { " + csps + "}"
+            cursor = self.db.aql.execute(query)
+            for doc in cursor:
+                g.add_node(doc['_id'], attr_dict = doc)
+        
+        for k,v in  self.cfg[gd.DB_SECTION][gd.DB_EDGE_COL].items():
+            query = "FOR doc in %s " % (k)
+            cspl = [s +':'+ 'doc.' + s for s in v]
+            cspl.append('_id: doc._id')
+            csps = ','.join(cspl)
+            query = query + "RETURN { " + csps + "}"
+            cursor = self.db.aql.execute(query)
+            #breakpoint()
+            for doc in cursor:
+                g.add_edge(doc['_from'], doc['_to'])
+        
+
+                        
```

Comment:
Lot of unnecessary whitespace, we should look into https://www.python.org/dev/peps/pep-0008/ style